skfuzzy/defuzzify/defuzz.py

```diff
@@ -58,7 +58,43 @@ def centroid(x, mfx):
     skfuzzy.defuzzify.defuzz, skfuzzy.defuzzify.dcentroid
 
     """
-    return (x * mfx).sum() / np.fmax(mfx.sum(),
+
+    '''
+    As we suppose linearity between each pair of points of x, we can calculate the exact area of the figure
+    (a triangle or a rectangle).
+    '''
+    sum_moment_area = 0.0
+    sum_area = 0.0
+    #If the membership function is a singleton fuzzy set:
+    if len(x)==1:
+        return x[0]*mfx[0]/np.fmax(mfx[0],
+                                     np.finfo(float).eps).astype(float)
+    #else return the sum of moment*area/sum of area
+    for i in range(1,len(x)):
```

Comment:
I can wrap it into a function and use the map instead of a for loop, and then use the `np.sum(, axis=0) ` to sum over the values. I do not know if map + np.sum would have a better performance than the for loop...what do you think?